Avoid playwright's buggy test.fail() skipping tests

lukaw3d opened the pull request:
A lot of playwright tests on master are being skipped, and CI doesn't fail
https://github.com/oasisprotocol/oasis-wallet-web/actions/runs/5826329626/job/15800051758#step:10:19
> ![image](https://github.com/oasisprotocol/oasis-wallet-web/assets/3758846/b20c4358-8970-4e18-a2ab-c421f57f0675)

This has been happening since https://github.com/oasisprotocol/oasis-wallet-web/pull/1570.
Additionally, https://github.com/oasisprotocol/oasis-wallet-web/pull/1583/commits/84b30f3c5270843d525ae8bb83dcea518fe572ad broke a test, and it wasn't caught.

Skipped tests seem to be an upstream issue. Reported to: https://github.com/microsoft/playwright/issues/26435

pro-wh commented on playwright/tests/fiat.spec.ts:
```diff
@@ -113,12 +126,11 @@
+      const rule = permissionsPolicy.find(rule => rule.startsWith(`${permission}=`))!
+      expect(
+        rule.endsWith('=*') ||
+          (rule.includes('https://global.transak.com') && rule.includes('https://global-stg.transak.com')),
```
rule is not a single URL right? should we suppress this?

lukaw3d replied:
Yeah, already suppressed to the best of my abilities. It's also just a test, testing our non-evil code :shrug:

pro-wh reviewed (commented):
those later tests are now run, but can this still report failing tests?

![image](https://github.com/oasisprotocol/oasis-wallet-web/assets/38229038/5d4e549d-9728-4c2b-a25c-615006253a8a)

![image](https://github.com/oasisprotocol/oasis-wallet-web/assets/38229038/83f3d06f-74b9-4a65-99e4-a6cc419a9001)

doperiddle commented:
Yes

On Tue, Aug 15, 2023, 10:43 AM pro-wh ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In playwright/tests/fiat.spec.ts
> <https://github.com/oasisprotocol/oasis-wallet-web/pull/1588#discussion_r1294919569>
> :
>
> > @@ -113,12 +126,11 @@
>      expect(transakPermissions).toBeTruthy()
>
>      for (const permission of transakPermissions!.split(';').map(permission => permission.trim())) {
> -      expect(permissionsPolicy.find(rule => rule.startsWith(`${permission}=`))).toContain(
> -        'https://global.transak.com',
> -      )
> -      expect(permissionsPolicy.find(rule => rule.startsWith(`${permission}=`))).toContain(
> -        'https://global-stg.transak.com',
> -      )
> +      const rule = permissionsPolicy.find(rule => rule.startsWith(`${permission}=`))!
> +      expect(
> +        rule.endsWith('=*') ||
> +          (rule.includes('https://global.transak.com') && rule.includes('https://global-stg.transak.com')),
>
> rule is not a single URL right? should we suppress this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/oasisprotocol/oasis-wallet-web/pull/1588#discussion_r1294919569>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHRNEHHBOLNSFRXZTPG4ZP3XVOYKPANCNFSM6AAAAAA3QHB77I>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>

lukaw3d commented:
Steps in https://github.com/oasisprotocol/oasis-wallet-web/pull/1588/commits/41da30749eebd7b333ec1e6da805915e0ffcf16d and https://github.com/oasisprotocol/oasis-wallet-web/pull/1588/commits/14f52305643d7ab01169a265b9ac7c0c88353de3
- split `beforeEach` into `expectNoErrorsInConsole(page); setup(page);`
- removed `expectNoErrorsInConsole` in tests with `test.fail`:
  Playwright skips when combining `test.fail` and assertions in event listeners (in `expectNoErrorsInConsole` there's: `page.on('pageerror', error => { expect(` and `page.on('console', message => { ... expect(`)
- added `page.waitForRequest` that gets blocked by security headers.
  Instead of finding errors in console.
- (when playwright is fixed we can re-add expectNoErrorsInConsole)
